spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/ManagedCommitSuite.scala

```diff
@@ -116,6 +123,405 @@ class ManagedCommitSuite
+      withTempDir { tempDir =>
+        val tablePath = tempDir.getAbsolutePath
+        val deltaLog1 = DeltaLog.forTable(spark, tablePath)
+        deltaLog1.startTransaction().commitManually(Metadata())
+        deltaLog1.startTransaction().commitManually(createTestAddFile("f1"))
+        deltaLog1.startTransaction().commitManually()
+        val snapshotV2 = deltaLog1.update()
+        assert(snapshotV2.version === 2)
+        assert(snapshotV2.commitStoreOpt.isEmpty)
+        DeltaLog.clearCache()
+
+        // Add new commit to convert FS table to managed-commit table
+        val deltaLog2 = DeltaLog.forTable(spark, tablePath)
+        val oldMetadata = snapshotV2.metadata
+        val newMetadata = snapshotV2.metadata.copy(
+          configuration =
+            oldMetadata.configuration + (MANAGED_COMMIT_OWNER_NAME.key -> "tracking-in-memory"))
+        deltaLog2.startTransaction().commitManually(newMetadata)
+        commitStore.registerTable(deltaLog2.logPath, 3)
+        deltaLog2.startTransaction().commitManually(createTestAddFile("f2"))
+        deltaLog2.startTransaction().commitManually()
+        val snapshotV5 = deltaLog2.unsafeVolatileSnapshot
+        assert(snapshotV5.version === 5)
+        assert(snapshotV5.commitStoreOpt.nonEmpty)
+        // only delta 4/5 will be un-backfilled and should have two dots in filename (x.uuid.json)
+        assert(snapshotV5.logSegment.deltas.count(_.getPath.getName.count(_ == '.') == 2) === 2)
+
+        val usageRecords = Log4jUsageLogger.track {
+          val newSnapshotV5 = deltaLog1.update()
+          assert(newSnapshotV5.version === 5)
+          assert(newSnapshotV5.logSegment.deltas === snapshotV5.logSegment.deltas)
+        }
+        assert(filterUsageRecords(usageRecords, "delta.readChecksum").size === 2)
+      }
+    } finally {
+      spark.conf.set(MANAGED_COMMIT_OWNER_NAME.defaultTablePropertyKey, oldCommitOwnerValue)
+    }
+  }
+
+  test("update works correctly with InitialSnapshot") {
+    CommitStoreProvider.registerBuilder(TrackingInMemoryCommitStoreBuilder(batchSize = 2))
+    withTempDir { tempDir =>
+      val tablePath = tempDir.getAbsolutePath
+      val clock = new ManualClock(System.currentTimeMillis())
+      val log = DeltaLog.forTable(spark, new Path(tablePath), clock)
+      assert(log.unsafeVolatileSnapshot.isInstanceOf[InitialSnapshot])
+      assert(log.getCapturedSnapshot().updateTimestamp == clock.getTimeMillis())
+      clock.advance(500)
+      log.update()
+      assert(log.unsafeVolatileSnapshot.isInstanceOf[InitialSnapshot])
+      assert(log.getCapturedSnapshot().updateTimestamp == clock.getTimeMillis())
+    }
+  }
+
+  // This test has the following setup:
+  // Setup:
+  // There are two table owners: CS1 and CS2 both of which are pointing to same underlying
+  // in-memory implementation.
+  // 1. Make 3 commits on the table with CS1 as owner.
+  // 2. Modify the content of commit-2. Change the table owner from 1 to 2 as part of it.
```

Comment:
AFAIK, the [RFC](https://github.com/delta-io/delta/blob/master/protocol_rfcs/managed-commits.md#converting-an-existing-filesystem-based-table-to-managed-commit-table) doesn't say anything about transferring a table between commit owners?

Reply:
We should add a section to RFC around Transferring commit ownership.

Reply:
The RFC already talks about FS -> MC and MC  -> FS. The same could be combined to change the owners.
i.e. MC1 -> FS -> MC2.

Reply:
After a bunch of discussion offline, it turns out direct ownership transfers are really messy, and hard to do securely, because they require two commit owners to coordinate in some ad-hoc way. At this point it looks better to keep the RFC spec as-is:
1. If the Delta client wants to propose an owner for a FS table, they must send a commit request to the proposed owner; if the owner agrees to take over the table, they arrange for a direct-backfilled commit that makes the owner change.
2. If the Delta client wants to remove the owner of a table (making it FS based), they must send one last commit request to the current owner; if the owner agrees to disown the table, they arrange for a direct-backfilled commit that makes the owner change.

The direct-backfilled commit may be done client-side by the owner's commit store, or server-side by the owner itself (that's up to the owner to decide).

Reply:
If we are in agreement about the above, we should update this test (and possibly others as well) to follow spec?